lib/Admin_Handler.cpp

```diff
@@ -322,6 +342,38 @@ bool FlushCommandWrapper(S* sess, const string& modname, char *query_no_space, i
 	return false;
 }
 
+std::tuple<bool, enum SERVER_TYPE, time_t>&& parse_command_purge_query_digests(char *query, int query_len) {
+	auto ret = new std::tuple<bool, enum SERVER_TYPE, time_t>(false, SERVER_TYPE_MYSQL, 0);
+
+	const char *prefix = match_command_prefix(CMD_PREFIX_PURGE_QUERY_DIGESTS, query, query_len);
+	if (prefix) {
+		// match_found
+		std::get<0>(*ret) = true;
+
+		// server_type
+		if (strstr(prefix, "_pgsql_") != nullptr) {
+			std::get<1>(*ret) = SERVER_TYPE_PGSQL;
+		}
+
+		// parse timestamp
+		char *timestamp = strdup(query + strlen(prefix));
+		int last_seen = atoi(trim_spaces_in_place(timestamp));
```

Comment:
`atoi` is used, risking overflow for large timestamps (e.g., post-2038).

Suggestion: 
* Replace `atoi `with `strtoll`.
* Change `last_seen` datatype to `time_t`

Reply:
Addressed.